modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i

```diff
@@ -0,0 +1,350 @@
+    h_name = hd
+    all_etas = 'etaa0 etab0 etad0'
+    phase_etas = 'etad0'
+  [../]
+  [./omegaa]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = omegaa
+    material_property_names = 'Vm ka caeq'
+    function = '-0.5*w^2/Vm^2/ka-w/Vm*caeq'
+    derivative_order = 2
+  [../]
+  [./omegab]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = omegab
+    material_property_names = 'Vm kb cbeq'
+    function = '-0.5*w^2/Vm^2/kb-w/Vm*cbeq'
+    derivative_order = 2
+  [../]
+  [./omegad]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = omegad
+    material_property_names = 'Vm kd cdeq'
+    function = '-0.5*w^2/Vm^2/kd-w/Vm*cdeq'
+    derivative_order = 2
+  [../]
+  [./rhoa]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = rhoa
+    material_property_names = 'Vm ka caeq'
+    function = 'w/Vm^2/ka + caeq/Vm'
+    derivative_order = 2
+  [../]
+  [./rhob]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = rhob
+    material_property_names = 'Vm kb cbeq'
+    function = 'w/Vm^2/kb + cbeq/Vm'
+    derivative_order = 2
+  [../]
+  [./rhod]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = rhod
+    material_property_names = 'Vm kd cdeq'
+    function = 'w/Vm^2/kd + cdeq/Vm'
+    derivative_order = 2
+  [../]
+  [./const]
+    type = GenericConstantMaterial
+    prop_names =  'kappa_c  kappa   L   D    Vm   ka    caeq kb    cbeq  kd    cdeq  gab gad gbd  mu  tgrad_corr_mult'
+    prop_values = '0        1       1.0 1.0  1.0  10.0  0.1  10.0  0.9   10.0  0.5   1.5 1.5 1.5  1.0 0.0'
+  [../]
+  [./Mobility]
+    type = DerivativeParsedMaterial
+    f_name = DchiVm
+    material_property_names = 'D chi Vm' #Factor of Vm is needed to evolve c instead of rho
+    function = 'D*chi*Vm'
+    derivative_order = 2
+  [../]
+  [./chi]
+    type = DerivativeParsedMaterial
+    f_name = chi
+    material_property_names = 'Vm ha(etaa0,etab0,etad0) ka hb(etaa0,etab0,etad0) kb hd(etaa0,etab0,etad0) kd'
+    function = '(ha/ka + hb/kb + hd/kd) / Vm^2'
+    args = 'etaa0 etab0 etad0'
+    derivative_order = 2
+  [../]
+  [./hoverk_a]
+    type = DerivativeParsedMaterial
+    material_property_names = 'ha(etaa0,etab0,etad0) Vm ka'
+    f_name = hoverk_a
+    function = 'ha / Vm / ka'
+  [../]
+  [./hoverk_b]
+    type = DerivativeParsedMaterial
+    material_property_names = 'hb(etaa0,etab0,etad0) Vm kb'
+    f_name = hoverk_b
+    function = 'hb / Vm / kb'
+  [../]
+  [./hoverk_d]
+    type = DerivativeParsedMaterial
+    material_property_names = 'hd(etaa0,etab0,etad0) Vm kd'
+    f_name = hoverk_d
+    function = 'hd / Vm / kd'
+  [../]
+[]
+
+[Preconditioning]
+  [./SMP]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Postprocessors]
+  [./c_total]
+    type = ElementIntegralVariablePostprocessor
+    variable = c
+  [../]
+[../]
+
+[Executioner]
+  type = Transient
+  nl_max_its = 15
+  scheme = bdf2
+  solve_type = PJFNK
```

Comment:
By the way this runs faster with `NEWTON`. You can then also drop the `[Preconditioning]` block as it gets added automatically for Newton solves.